### PR TITLE
Modify somali is_noise to return False if any other cleaner matches

### PR DESCRIPTION
**Pull request by as2388**

### CODE REVIEW

File: core_data_modules/cleaners/somali/demographic_cleaner.py [hunk 1]
```diff
@@ -3,8 +3,18 @@
 
 
 class DemographicCleaner(object):
-    @staticmethod
-    def is_noise(text):
+    @classmethod
+    def is_noise(cls, text):
```
Comment (lukechurch) [inline core_data_modules/cleaners/somali/demographic_cleaner.py]: I'm a little uncomfortable about the name - this is more detecting 'is this a possible demographic string', not noise independent of context. Any thoughts?

File: tests/cleaners/somali/test_demographic_cleaner.py [hunk 2]
```diff
@@ -7,12 +7,27 @@
...
+    def test_clean_is_only_yes_no(self):
+        self.assertTrue(DemographicCleaner.is_only_yes_no("ha"))
+        self.assertTrue(DemographicCleaner.is_only_yes_no("my"))
+
     def test_clean_genders(self):
         self.assertEqual(DemographicCleaner.clean_gender("woman"), Codes.FEMALE)
 
     def test_clean_yes_no(self):
         self.assertEqual(DemographicCleaner.clean_yes_no("haa"), Codes.YES)
+        # self.assertEqual(DemographicCleaner.clean_yes_no("ha"), Codes.YES)  # TODO: Test fails
```
Comment (lukechurch) [inline tests/cleaners/somali/test_demographic_cleaner.py]: Issues please

File: core_data_modules/cleaners/somali/demographic_cleaner.py (hunk 1 above)
Comment (lukechurch) [reply core_data_modules/cleaners/somali/demographic_cleaner.py]: Consider 'is_possible_demographic_answer'

Comment (as2388) [reply core_data_modules/cleaners/somali/demographic_cleaner.py]: It would actually be `is_not_possible_demographic_answer`

Comment (as2388) [reply core_data_modules/cleaners/somali/demographic_cleaner.py]: Hmm. The problem is the noise regex is still looking for specific patterns known to be noise. The changes here are to reduce the number of occasions where that filter matches demographics.

Comment (as2388) [reply core_data_modules/cleaners/somali/demographic_cleaner.py]: Maybe we should break this into two functions?

File: tests/cleaners/somali/test_demographic_cleaner.py (hunk 2 above)
Comment (as2388) [reply tests/cleaners/somali/test_demographic_cleaner.py]: https://github.com/AfricasVoices/CoreDataModules/issues/61

File: core_data_modules/cleaners/somali/demographic_cleaner.py (hunk 1 above)
Comment (as2388) [reply core_data_modules/cleaners/somali/demographic_cleaner.py]: Although then we need to encourage users to call both, which doesn't feel great

Comment (lukechurch) [reply core_data_modules/cleaners/somali/demographic_cleaner.py]: Should we split into two functions 'pre_filter_...' and 'is_noise_by_regex' and then have a 'is_noise' method that calls one and then the other?

Comment (as2388) [reply core_data_modules/cleaners/somali/demographic_cleaner.py]: Will keep the name, but re-structure to make it clear that this is a two-phase process

**Pull request merged**